src/macros.jl

```diff
@@ -523,6 +604,50 @@ end
 ### Build constraints using actual data.
 ###
 
+function build_constraint(
+    _error::Function,
+    f,
+    set::Nonnegatives,
+    args...;
+    kwargs...,
+)
+    return build_constraint(_error, f, MOI.GreaterThan(0.0), args...; kwargs...)
+end
+
+function build_constraint(
+    _error::Function,
+    f,
+    set::Nonpositives,
+    args...;
+    kwargs...,
+)
+    return build_constraint(_error, f, MOI.LessThan(0.0), args...; kwargs...)
+end
+
+function build_constraint(_error::Function, f, set::Zeros, args...; kwargs...)
+    return build_constraint(_error, f, MOI.EqualTo(0.0), args...; kwargs...)
+end
+
+function build_constraint(
+    _error::Function,
+    f::AbstractVector,
+    set::Nonnegatives,
+)
+    return build_constraint(_error, f, MOI.Nonnegatives(length(f)))
```

Comment:
This is equivalent to assuming that `@constraint(model, a >= b)` is equivalent to `@constraint(model, a >= b, Nonnegatives())` right ?
I think we explicitly decided not to have these cones by default but I would be ok to change it but I think we need to make sure everybody agrees on it.
It also kind of makes sense since we assumes `Nonnegatives` by default to compare vector of objective in multiobjective optimization.

Reply:
> This is equivalent to assuming that @constraint(model, a >= b) is equivalent to @constraint(model, a >= b, Nonnegatives()) right?

Yes, for vector-valued `a - b`. I think this is pretty standard interpretation. Do you remember why we decided not to?

It makes it a lot easier to write models that conic solvers support without bridges. Currently they'll bridge _every_ scalar linear constraint to a vector-valued one of dimension 1.

Reply:
I don't remember, @mlubin do you ?
It's also the convention Boyd's uses in his book.